File: field/contests/atcoder/abc339/b.py
import sys
input = lambda: sys.stdin.readline().rstrip()
INF = 10**18

H,W,N = map(int,input().split())
grid = [["."]*W for _ in range(H)]

cy,cx = 0,0
dir = "U"
DIR = {"U":(0,1), "R":(1,0), "D":(0,-1), "L":(-1,0)}
DIR2 = {"U":"R", "R":"D", "D":"L", "L":"U"}
DIR3 = {"U":"L", "R":"U", "D":"R", "L":"D"}

for i in range(N):
    if grid[cy][cx] == ".":
        grid[cy][cx] = "#"
        cy,cx = (cy+DIR[dir][0])%H, (cx+DIR[dir][1])%W
        dir = DIR2[dir]
    else:
        grid[cy][cx] = "."
        cy,cx = (cy-DIR[dir][0])%H, (cx-DIR[dir][1])%W
        dir = DIR3[dir]

for i in range(H):
    print("".join(grid[i]))

File: field/contests/atcoder/abc339/src/bin/b.rs
#![allow(dead_code,unused_imports,unused_variables,non_snake_case, non_upper_case_globals, path_statements)]
use proconio::{fastout, input,marker::{Chars, Bytes, Isize1, Usize1}};
use std::{
    cmp::{max, min, Reverse},
    collections::{BTreeMap, BTreeSet, BinaryHeap, HashMap, HashSet, VecDeque},
    mem::swap,
    process::exit,
};
use itertools::Itertools;
use superslice::Ext;

const INF: usize = 1 << 60;

const MOVE: [(usize, usize); 4] = [(!0, 0), (0, 1), (1, 0), (0, !0)]; //縦横移動

#[fastout]
fn main() {
    input! {
        H: usize,
        W: usize,
        N: usize,
    }

    let mut grid = vec![vec!['.';W];H];
    let mut dir = 0;
    let (mut cy, mut cx) = (0,0);
    for _ in 0..N {
        if grid[cy][cx] == '.' {
            grid[cy][cx] = '#';
            dir = (dir + 1) % 4;
        } else {
            grid[cy][cx] = '.';
            dir = (dir + 3) % 4;
        }
        let (dy, dx) = MOVE[dir];
        cy = (cy + dy + H) % H;
        cx = (cx + dx + W) % W;
    }
    println!("{}",grid.iter().map(|x| x.iter().join("")).join("\n"));
}
